### DISCUSSION TITLE
DTO record 사용 어떠실까요?

### DISCUSSION THREAD

**Discussion by kmw2378** [KakaoFunding/back-end#23] [Q&A] [answered]
DTO 작성 시 레코드를 사용하는 건 어떨까요?

장점은 아래와 같이 DTO 작성 코드가 간단해집니다.
```java
public record OAuthAuthenticateRequest(String provider, String code) {
}
```

단점은 Getter 메서드명이 get + (필드명) 이 아닌 `필드명()` 형태라 혼동이 있습니다.
e.g. MemberDto의 `name` 필드
```java
MemberDto memberDto = new MemberDto(...);
memberDto.getName(); // X
memberDto.name();  // O
```

**Comment by YeaChan05** [comment, marked as answer]
마침 오늘[ 관련된 포스팅](https://ducktopia.tistory.com/m/106)을 봤는데 우연이네요 ㅎㅎ
단순히 데이터 전달용으로 편하게 쓸때는 확실히 용이해보이지만 record가 상송도 불가능하고(뭐 dto에 얼마나 쓸일이 있겠냐만은...) 결국 `@Builder`나 `@ToString`같은 Lombok에 유용한 기능들도 dto에 필요한 경우가 많아서 고민이 되긴 하군요

**Reply by YeaChan05** [reply]
방금 댓글 달면서 알게된건데 record의 경우에는 Lombok에서[ `@RecordBuilder`를 따로 지원해준다고 하네요!](https://howtodoinjava.com/java/basics/builder-pattern-for-java-records/)
확실히 자바 11버전보다 상위 버전을 쓴다면 그에 맞는 자바 기능들을 적극적으로 활용하는것도 방법이겠네요

**Reply by YeaChan05** [reply]
적극 찬성합니다 😁😁

**Reply by kmw2378** [reply]
오오 `@RecordBuilder` 는 처음 보네요! 말씀하신대로 11 이후 버전에서 제공해주는 기능을 적극적으로 쓰는게 좋다고 생각합니다! 좋은 정보 정말 감사합니다 ㅎㅎ
